src/networking.c

```diff
@@ -714,6 +714,14 @@ void setDeferredAggregateLen(client *c, void *node, long length, char prefix) {
      * we return NULL in addReplyDeferredLen() */
     if (node == NULL) return;
 
+    /* Things like *2\r\n or *3\r\n are emitted very often by the protocol
+     * so we have a few shared objects to use if the integer is small
+     * like it is most of the times. */
+    if (prefix == '*' && length < OBJ_SHARED_BULKHDR_LEN) {
+        setDeferredReply(c, node, shared.mbulkhdr[length]->ptr, (length < 10) ? 4 : 5);
```

Comment:
you're making some assumptions on OBJ_SHARED_BULKHDR_LEN
```suggestion
        setDeferredReply(c, node, shared.mbulkhdr[length]->ptr, sdslen(shared.mbulkhdr[length]->ptr));
```
considering the sds will get to the CPU cache anyway, i don't think this will have any performance impact

Reply:
@oranagra the sdslen change makes:
- pipeline 16: go from 880K to 875K ops/sec. ~0.4% performance drop
- pipeline 1: --no measurable change--

should we go with the cleaner sdslen or revert to use the conditional?

Reply:
are you sure it's consistent?
we can also add some static asserts and such, or a macro to get the length that's declared next to where OBJ_SHARED_BULKHDR_LEN is declared, but it seems like the wrong thing to do.

Reply:
@oranagra can you confirm https://github.com/redis/redis/pull/10334/files#diff-e90a5ffcc23440393815655ee66ce9301d183a681056ee8020e39d5df2c4fa0fR112 solves it.